cmd/set.go

```diff
@@ -45,48 +53,126 @@ func init() {
+		fmt.Printf("\nCredential %s has been set at %s/%s\n", name, pe, name)
+	}
 
 	hints.Display(hints.View, hints.Run)
 	return nil
 }
 
-func parseSetArgs(args []string) (key string, value string, errMsg string) {
-	// If there is only one argument, try to parse using env var syntax: KEY=value
+// parseSetSecrets returns a list of secret pairs either from the command line
+// arguments or from the standard input. It returns an error if there is a
+// problem parsing secrets or stdin fails to read.
+func parseSetSecrets(args []string) ([]secretPair, error) {
+	if len(args) > 0 {
+		secret, err := parseSetArgs(args)
+		return []secretPair{secret}, err
+	}
+
+	// If no arguments were passed, try to read from stdin.
+	input, err := readStdin()
+	if err != nil {
+		return nil, err
+	}
+
+	return parseSetStdin(input)
+}
+
+// parseSetArgs returns a secret pair from a list of arguments. If there is only
+// one argument, try to parse using env var syntax: `KEY=VALUE`.
+func parseSetArgs(args []string) (pair secretPair, err error) {
 	if len(args) == 1 {
 		args = strings.SplitN(args[0], "=", 2)
 	}
 
-	switch len(args) {
-	case 0, 1:
-		errMsg = "A secret name and value must be supplied."
-	case 2:
-		key = args[0]
-		value = args[1]
+	if len(args) < 2 {
+		return pair, errors.New("A secret name and value must be supplied")
+	} else if len(args) > 2 {
+		return pair, errors.New("Too many arguments were provided")
+	}
+
+	pair.key = args[0]
+	pair.value = args[1]
 
-		if key == "" || value == "" {
-			errMsg = "A secret must have a name and value."
+	if pair.key == "" || pair.value == "" {
+		return pair, errors.New("A secret must have a name and value")
+	}
+
+	return pair, nil
+}
+
+// parseSetStdin reads from the standard input parsing each line as a possible
+// secret pair with `KEY=VALUE` format. Spaces around `=` are discarded.
+func parseSetStdin(lines []string) ([]secretPair, error) {
+	var pairs []secretPair
+
+	for i, line := range lines {
+		args := strings.SplitN(line, "=", 2)
+		if len(args) < 2 {
+			return nil, fmt.Errorf("Error parsing line %d: %q", i+1, line)
+		}
+
+		pair := secretPair{
+			key:   strings.Trim(args[0], " "),
+			value: strings.Trim(args[1], " "),
+		}
+
+		if pair.key == "" || pair.value == "" {
+			return nil, fmt.Errorf("Error parsing line %d: %q", i+1, line)
 		}
-	default:
-		errMsg = "Too many arguments were provided."
+
+		pairs = append(pairs, pair)
+	}
+
+	return pairs, nil
+}
+
+// readStdin checks if stdin is available and can be read. It returns a list of
+// non-empty lines.
+func readStdin() ([]string, error) {
+	stat, err := os.Stdin.Stat()
+	if err != nil {
+		return nil, fmt.Errorf("Could not from stdin. %s", err)
+	}
+
+	if (stat.Mode() & os.ModeNamedPipe) != 0 {
+		return nil, errors.New("Could not read from piped input")
+	}
+
+	var lines []string
+
+	scanner := bufio.NewScanner(os.Stdin)
+	for scanner.Scan() {
+		line := scanner.Text()
+
+		if line != "" {
+			lines = append(lines, line)
+		}
+	}
```

Comment:
It would be very cool if you incorporated something like https://github.com/google/shlex into this, treating the contents as a shell style .env file. Then comments and quoting/escaping would be handled in the same way as other tools handle env configuration in files.